pypeit/spectrographs/mdm_modspec.py

```diff
@@ -0,0 +1,284 @@
+        par['calibrations']['biasframe']['exprng'] = [None, 1]
+        par['calibrations']['darkframe']['exprng'] = [999999, None]     # No dark frames
+        par['calibrations']['pinholeframe']['exprng'] = [999999, None]  # No pinhole frames
+        par['calibrations']['arcframe']['exprng'] = [None, None]  # Long arc exposures on this telescope
+        par['scienceframe']['exprng'] = [10, 600]
+
+        return par
+
+    def init_meta(self):
+        """
+        Define how metadata are derived from the spectrograph files.
+
+        That is, this associates the ``PypeIt``-specific metadata keywords
+        with the instrument-specific header cards using :attr:`meta`.
+        """
+        # see https://pypeit.readthedocs.io/en/latest/setup.html?highlight=decker#overview
+        self.meta = {}
+        # Required (core)
+        self.meta['ra'] = dict(ext=0, card='RA')
+        self.meta['dec'] = dict(ext=0, card='DEC')
+        self.meta['target'] = dict(ext=0, card='OBJECT')
+        self.meta['decker'] = dict(card=None, compound=True)
+        self.meta['binning'] = dict(card=None, compound=True)
+        
+        #self.meta['datasec'] = dict(ext=0, card='DATASEC') # possibly use local variable for this
+        self.meta['filter1'] = dict(ext=0, card='FILTER') 
+
+        
+        self.meta['mjd'] = dict(card=None, compound=True)
+        self.meta['exptime'] = dict(ext=0, card='EXPTIME')
+        self.meta['airmass'] = dict(ext=0, card='AIRMASS')
+        
+        # Extras for config and frametyping
+        self.meta['dispname'] = dict(card=None, compound=True) 
+        self.meta['idname'] = dict(ext=0, card='IMAGETYP')
+        self.meta['cenwave'] = dict(card=None, compound=True, rtol=2.0)
+
+       
+        # Lamps
+        self.meta['lampstat01'] = dict(ext=0, card='LAMPS')
+        self.meta['instrument'] = dict(ext=0, card='INSTRUME')
+        
+        # Mirror
+        self.meta['mirror'] = dict(ext=0, card='MIRROR')
+        
+    def compound_meta(self, headarr, meta_key):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        if meta_key == 'binning':
+            binspatial = headarr[0]['CCDBIN1']
+            binspec = headarr[0]['CCDBIN2']
+            return parse.binning2string(binspec, binspatial)
+        if meta_key == 'mjd':
+            return Time(headarr[0]['JD'], format='jd').mjd
+        if meta_key == 'decker':
+            return 'none'
+        if meta_key == 'dispname':
+            return '1200 l/mm'
+        if meta_key == 'cenwave':
+            return 5100.0
```

Comment:
Is this always true?  If not, does it matter?

Reply:
not sure;  leaving for now..